File: 111_MinimumDepthOfBinaryTree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
//18.75%
int minDepth(struct TreeNode* root) {
    if(!root)
		return 0;
	if(!root -> left)
		return minDepth(root -> right) + 1;
	if(!root -> right)
		return minDepth(root -> left) + 1;
	
	int heightL = minDepth(root -> left);
	int heightR = minDepth(root -> right);
	return  heightL < heightR ? heightL + 1 : heightR + 1;
}

File: 111_MinimumDepthOfBinaryTree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None
# 28.81%
class Solution(object):
    def minDepth(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        if not root:
            return 0
        if not root.left:
            return self.minDepth(root.right) + 1
        if not root.right:
            return self.minDepth(root.left) + 1
        return min(self.minDepth(root.right), self.minDepth(root.left)) + 1

# From LeetCode		
def minDepth(self, root):
    if not root: return 0
    d = map(self.minDepth, (root.left, root.right))
    return 1 + (min(d) or max(d))
	

def minDepth(self, root):
    if not root: return 0
    d, D = sorted(map(self.minDepth, (root.left, root.right)))
    return 1 + (d or D)